airgun/views/hardware_model.py

```diff
@@ -11,7 +11,7 @@ class HardwareModelsView(BaseLoggedInView, SearchableViewMixin):
     table = SatTable(
         './/table',
         column_widgets={
-            'Name': Text('./a'),
+            'Name': Text('.//a[contains(@href, "/edit")]'),
```

Comment:
Couldn't that be just `.//a`? Do we expect more than one `<a>` in this cell, so we need to be more specific here?

Reply:
I wanted to be more specific, but ".//a" would work as well. Is that (to be as generic as possible) generic rule? Just out of interest: what is the reasoning behind that?

Reply:
As a general rule, we need to find right balance between:

* stability - small changes in product should not require large changes in framework (ideally, small changes in product would not require any changes in framework)
* specificity - locator should find only the element we need
* easiness to read and maintain - length is quite good proxy for these

Of course there is a lot of personal judgement involved. Like in your case - it's not your locator is bad (that's why I acked this change), I just felt simpler one could suffice.

Reply:
Ack, I have simplified it. Thank you!